Contents/Code/siteUltrafilms.py

```diff
@@ -0,0 +1,147 @@
+    maxscore = 0
+
+    try:
+        searchResults = HTML.ElementFromURL(url)
+    except:
+        request = urllib.Request(url, headers=headers)
+        response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
+        htmlstring = response.read()
+        searchResults = fromstring(htmlstring)
+
+    for searchResult in searchResults.xpath('//ul[@class="listing-videos listing-tube"]/*[div[@class="format-infos"]/div[@class="time-infos"]]'):
+        titleNoFormatting = searchResult.xpath('.//a')[0].get('title').strip()
+        Log("titleNoFormatting: " + titleNoFormatting)
+        actors = searchResult.xpath('.//img')[0].get('alt')
+        Log(actors)
+        sceneUrl = searchResult.xpath('.//a')[0].get('href')
+        curID = sceneUrl.replace('/','_').replace('?','!')
+        Log("curID: " + curID)
+        score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
+        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " with "+ actors + " [" + PAsearchSites.getSearchSiteName(siteNum) + "] ", score = score, lang = lang))
+        if maxscore < score:
+            maxscore = score
+
+        Log("MaxScore:" + str(maxscore))
+
+    if maxscore < 100:
+        Log("no exact match, trying with normal search")
+        url = PAsearchSites.getSearchSearchURL(siteNum) + encodedTitle
+        headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'}
+
+        try:
+            searchResults = HTML.ElementFromURL(url)
+        except:
+            request = urllib.Request(url, headers=headers)
+            response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
+            htmlstring = response.read()
+            searchResults = fromstring(htmlstring)
+
+        for searchResult in searchResults.xpath('//ul[@class="listing-videos listing-tube"]/*[div[@class="format-infos"]/div[@class="time-infos"]]'):
+            titleNoFormatting = searchResult.xpath('.//a')[0].get('title').strip()
+            Log("titleNoFormatting: " + titleNoFormatting)
+            actors = searchResult.xpath('.//img')[0].get('alt')
+            Log(actors)
+            sceneUrl = searchResult.xpath('.//a')[0].get('href')
+            curID = sceneUrl.replace('/','_').replace('?','!')
+            Log("curID: " + curID)
+            score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
+            results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " with "+ actors + " [" + PAsearchSites.getSearchSiteName(siteNum) + "] ", score = score, lang = lang))
+
+    return results
+        
+
+def update(metadata,siteID,movieGenres,movieActors):
+    pageURL = str(metadata.id).split("|")[0].replace('_', '/').replace('?','!')
```

Comment:
Could you throw in a `Log('******UPDATE CALLED*******')` right below the `def update`?